### PR TITLE
Update notification.lua

### PR DESCRIPTION
**Pull request by OmegaExtern**
Fixes overlapping bug in AddProgress function.

### CODE REVIEW

**Comment by PotcFdk** [comment]
I object to this bugfix plus whitespacing-adjustment nightmare.

**Comment by Kefta** [comment]
@PotcFdk Why?

**Comment by OmegaExtern** [comment]
@PotcFdk Reason?
Anyway, think again.

First:
notification.AddProgress( "SomeID", "Some text" )
https://i.imgur.com/dZ9hRAs.png

Second:
notification.AddProgress( "AnotherID", "Another text" )
https://i.imgur.com/rSnxcn9.png

Third:
notification.AddProgress( "SomeID", "New text" )
https://i.imgur.com/NFVZIll.png

Also, without this bug-fix it is impossible to get rid off the first notification (using [notification.Kill](http://wiki.garrysmod.com/page/notification/Kill)), because the first notification gets replaced by the third notification.
And yes, this patch also formats notification.lua file with a style that is suggested on Garry's Mod wikipedia (notice it is also used widely in this repository).

**Comment by PotcFdk** [comment]
@OmegaExtern Because this is an ugly PR. All your changes are in one huge commit, including whitespacing changes. It's really hard and annoying to find out what is going on.

http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/

**Comment by Kefta** [comment]
@PotcFdk You can easily tell the changes, and less commits in this case is fine for a chained change like this. The white space can be fixed afterwards if the author doesn't fix it himself

**Comment by Kefta** [comment]
A change shouldn't be discouraged because of a few cosmetic things that can easily be fixed by anyone afterward

**Comment by LennyPenny** [comment]
no because then we'd have a cleanup commit after every contributed commit lol

**Comment by Kefta** [comment]
It's just one commit; this hasn't been an issue in the padt

**Comment by GGG-KILLER** [comment]
"Oh, this code does not follow my coding style standards, let's just reject it because all these spaces are confusing me"
If it adds a functionality, it works and the code can be understood, then why not?
All these whitespaces in fact made the code more readable in my opinion.

File: garrysmod/lua/includes/modules/notification.lua [hunk 1]
```diff
@@ -207,35 +209,33 @@ function PANEL:SetLegacyType( t )
...
-		
+
 		surface.SetDrawColor( 0, 50, 0, 255 )
 		surface.DrawRect( 5, self:GetTall() - 9, self:GetWide() - 10, 3 )
-		
+
 		local w = self:GetWide() * 0.25
 		local x = math.fmod( SysTime() * 200, self:GetWide() + w ) - w
-		
-		if ( x + w > self:GetWide() - 11 ) then w = ( self:GetWide() - 11 ) - x end
```
Comment (OmegaExtern) [inline garrysmod/lua/includes/modules/notification.lua]: Removed a whitespace at the end of the line - not entire line.

File: garrysmod/lua/includes/modules/notification.lua [hunk 2]
```diff
@@ -79,45 +89,40 @@ local function UpdateNotice( i, Panel, Count )
...
-	
+
 	y = y + Panel.VelY * spd
 	x = x + Panel.VelX * spd
-	
+
 	local dist = ideal_y - y
 	Panel.VelY = Panel.VelY + dist * spd * 1
-	if (math.abs(dist) < 2 && math.abs(Panel.VelY) < 0.1) then Panel.VelY = 0 end
-	local dist = ideal_x - x
```
Comment (OmegaExtern) [inline garrysmod/lua/includes/modules/notification.lua]: Notice "dist" is already defined as local variable above.

File: garrysmod/lua/includes/modules/notification.lua [hunk 3]
```diff
@@ -79,45 +89,40 @@ local function UpdateNotice( i, Panel, Count )
 
 	local x = Panel.fx
 	local y = Panel.fy
-	
-	local w = Panel:GetWide()
-	local h = Panel:GetTall()
-
-	w = w + 16
-	h = h + 16
```
Comment (OmegaExtern) [inline garrysmod/lua/includes/modules/notification.lua]: Added on initialization line (same for "w").

File: garrysmod/lua/includes/modules/notification.lua [hunk 4]
```diff
@@ -25,6 +25,16 @@ local Notices = {}
 
 function AddProgress( uid, text )
 
+	if ( IsValid( Notices[ uid ] ) ) then
```
Comment (OmegaExtern) [inline garrysmod/lua/includes/modules/notification.lua]: Actual bug fix.

**Comment by PotcFdk** [comment]
I'm not saying that changes to the whitespacing shouldn't happen in any case. But jesus, please, at least split them into their own commit so the owners/contributors can cherry-pick if necessary.
I'm not trying to just put blames into this PR - it's just that the use of Git comes with some *common knowledge* / Dos and Donts.

**Comment by Bo98** [comment]
Regarding the:

> It's really hard and annoying to find out what is going on.

That's solvable with the `?w=1` URL parameter:

https://github.com/garrynewman/garrysmod/pull/1075/files?w=1

**Comment by PotcFdk** [comment]
That isn't available in Git, or just outside of GitHub at all - so it's an ugly service-dependant solution to an unnecessary issue. Stop avoiding the fact.

**Comment by GGG-KILLER** [comment]
> That isn't available in Git, or just outside of GitHub at all - so it's an ugly service-dependant solution to an unnecessary issue. Stop avoiding the fact.

This is GitHub and yes, we will use GitHub-specific solutions.
Proper use of git was never required for pull requests, so just stop complaining about the way he did the pull request and instead give some actual feedback on what matters: *the code*

**Pull request merged**